tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py

```diff
@@ -96,3 +98,332 @@ def test_label_swirl(self, xp, scp):
         x = xp.array(x)
         labels, num_features = scp.ndimage.label(x)
         return labels
+
+
+@testing.gpu
+@testing.with_requires('scipy')
+class TestNdimageSum(unittest.TestCase):
+
+    @testing.for_dtypes([cupy.int32, cupy.float32, cupy.float64, cupy.uint32,
+                         cupy.uint64, cupy.ulonglong, cupy.int64, cupy.uint8,
+                         cupy.uint16])
+    @testing.numpy_cupy_array_almost_equal(scipy_name='scp')
```

Comment:
You may use `@testing.numpy_cupy_array_equal` after changing the output dtype of `sum`, removing `astype` in the return statement.